packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -901,10 +899,7 @@ export const DatePickerMixin = (subclass) =>
     _onChange(event) {
       // For change event on the native <input> blur, after the input is cleared,
       // we schedule change event to be dispatched on date-picker blur.
-      if (
-        this.inputElement.value === '' &&
-        !(event.detail && event.detail.sourceEvent && event.detail.sourceEvent.__fromClearButton)
```

Comment:
We no longer use `__fromClearButton`, this logic was needed when using `vaadin-text-field` (before Vaadin 22).